File: daily-hello-service/internal/models/device.go
package models

import "time"

type DeviceStatus string

const (
	DeviceStatusPending  DeviceStatus = "pending"
	DeviceStatusApproved DeviceStatus = "approved"
	DeviceStatusRejected DeviceStatus = "rejected"
)

type Device struct {
	ID         uint         `json:"id" gorm:"primaryKey"`
	UserID     *uint        `json:"user_id" gorm:"index"`
	User       *User        `json:"user,omitempty" gorm:"foreignKey:UserID"`
	DeviceID   string       `json:"device_id" gorm:"type:varchar(100)"`
	DeviceName string       `json:"device_name" gorm:"type:varchar(200)"`
	Platform   string       `json:"platform" gorm:"type:varchar(20)"`
	Model      string       `json:"model" gorm:"type:varchar(100)"`
	Status     DeviceStatus `json:"status" gorm:"type:varchar(20);default:'pending'"`
	ApprovedBy *uint        `json:"approved_by,omitempty" gorm:"index"`
	ApprovedAt *time.Time   `json:"approved_at,omitempty"`
	CreatedAt  time.Time    `json:"created_at"`
}

type RegisterDeviceRequest struct {
	DeviceID   string `json:"device_id" validate:"required"`
	DeviceName string `json:"device_name"`
	Platform   string `json:"platform"`
	Model      string `json:"model"`
}

type DeviceStatusQuery struct {
	DeviceID string `query:"device_id" validate:"required"`
}

type DeviceListQuery struct {
	Status string `query:"status"`
}

File: daily-hello-app/lib/models/device.dart
class Device {
  final int id;
  final int? userId;
  final String deviceId;
  final String deviceName;
  final String platform;
  final String model;
  final String status; // pending, approved, rejected
  final int? approvedBy;
  final DateTime? approvedAt;
  final DateTime createdAt;

  Device({
    required this.id,
    this.userId,
    required this.deviceId,
    required this.deviceName,
    required this.platform,
    required this.model,
    required this.status,
    this.approvedBy,
    this.approvedAt,
    required this.createdAt,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: (json['id'] as num).toInt(),
      userId: (json['user_id'] as num?)?.toInt(),
      deviceId: json['device_id'] ?? '',
      deviceName: json['device_name'] ?? '',
      platform: json['platform'] ?? '',
      model: json['model'] ?? '',
      status: json['status'] ?? 'pending',
      approvedBy: (json['approved_by'] as num?)?.toInt(),
      approvedAt: json['approved_at'] != null
          ? DateTime.parse(json['approved_at'] as String)
          : null,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  bool get isPending => status == 'pending';
  bool get isApproved => status == 'approved';
  bool get isRejected => status == 'rejected';
}
